src/App.js

```diff
@@ -1,79 +1,127 @@
 import React from 'react';
 
-import { getAll, getById } from './api/phone'
+import {getAll, getById} from './api/phone'
 import Basket from './Basket'
 import Filter from './Filter'
 import Catalog from './Catalog'
-
+import Viewer from './Viewer'
 import './App.css';
 
 
 class App extends React.Component {
-  constructor(props) {
-    super(props);
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            phones: getAll(),
+            selectedPhone: null,
+            basketItems: [],
+        };
+    }
+
+    addToBasket = (phoneId) => {
+        const items = this.state.basketItems;
+        const isItemPresent = items.filter(el => el.name === phoneId).length > 0;
+        if (isItemPresent) {
+            let item = items.find(el => {
+                return el.name === phoneId;
+            });
+            item.count = Number.parseInt(item.count) + 1;
 
-    this.state = {
-      phones: getAll(),
-      selectedPhone: null,
-      basketItems: [],
+            this.setState({
+                basketItems: [...items]
+            })
+        }
+        else {
+            this.setState({
+                basketItems: [...items, {name: phoneId, count: 1}]
+            })
+        }
+    };
+    removeFromBasket = (index) => {
+        console.log(index);
+        let newBasket = this.state.basketItems;
+        newBasket.splice(index);
+        this.setState({
+            basketItems: newBasket
+        })
+    };
+    sortPhones = (e) => {
+        switch (e) {
+            case 'name':
+                this.sortByName();
+                break;
+            case 'age':
+                this.sortByAge();
+                break;
+            default:
+                this.sortByName();
+        }
+    };
+    sortByName = () => {
+        this.setState({
+            phones: getAll(),
```

Comment:
```suggestion
            phones: [...prevState.phones].sort((a, b) => a.name.localeCompare(b.name)),
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare